idaes/power_generation/unit_models/waterpipe.py

```diff
@@ -0,0 +1,404 @@
+
+        self.control_volume.add_state_blocks(has_phase_equilibrium=False)
+
+        self.control_volume.add_material_balances(
+            balance_type=self.config.material_balance_type,
+            )
+
+        self.control_volume.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_transfer=self.config.has_heat_transfer)
+
+        self.control_volume.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=True)
+
+        # Add Ports
+        self.add_inlet_port()
+        self.add_outlet_port()
+
+        # Add object references
+        self.volume = Reference(self.control_volume.volume)
+
+        # Set references to balance terms at unit level
+        if (self.config.has_heat_transfer is True and
+                self.config.energy_balance_type != EnergyBalanceType.none):
+            self.heat_duty = Reference(self.control_volume.heat)
+
+        if (self.config.has_pressure_change is True and
+                self.config.momentum_balance_type != 'none'):
+            self.deltaP = Reference(self.control_volume.deltaP)
+
+        # Set Unit Geometry and Volume
+        self._set_geometry()
+
+        # Construct performance equations
+        self._make_performance()
+
+    def _set_geometry(self):
+        """
+        Define the geometry of the unit as necessary
+        """
+        # Number of pipe
+        self.number_of_pipes = Var(
+                initialize=4,
+                doc="Number of pipes")
+        # Length of pipe
+        self.length = Var(
+                initialize=10.0,
+                doc="Total length of the straight pipe")
+        # Elevation change of pipe, elevation of outlet - elevation of inlet
+        self.elevation_change = Var(
+                initialize=10.0,
+                doc="Change of elevation from inlet to outlet")
+        # Inside diameter of pipe
+        self.diameter = Var(
+                initialize=0.6,
+                doc="Inside diameter of pipe")
+
+        if not self.config.contraction_expansion_at_end == "None":
+            self.area_ratio = Var(
+                initialize=1,
+                doc="Cross section area ratio of exit end to pipe")
+
+        # Volume constraint
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Total volume of all pipes")
+        def volume_eqn(b, t):
+            return b.volume[t] == 0.25 * const.pi * b.diameter**2 \
+                * b.length * b.number_of_pipes
+
+    def _make_performance(self):
+        """
+        Define constraints which describe the behaviour of the unit model.
+        """
+        phase = self.config.water_phase
+
+        # Add performance variables
+        # Velocity of fluid inside pipe
+        self.velocity = Var(
+                self.flowsheet().config.time,
+                initialize=10.0,
+                doc='Liquid water velocity inside pipe')
```

Comment:
Given we also support vapor, should this be "Fluid"  or just "Water" instead of "Liquid water"?